File: .trae/skills/frontend-backend-test/scripts/ui-test.py
#!/usr/bin/env python3
"""
UI测试脚本
用于测试前端页面的功能和用户体验
"""

import os
import time
from playwright.sync_api import sync_playwright

# 前端URL
FRONTEND_URL = "http://localhost:3001"

# 测试订单页面
def test_order_page():
    """测试订单页面"""
    print("开始UI测试...")
    
    with sync_playwright() as p:
        # 启动浏览器
        browser = p.chromium.launch(headless=False)
        page = browser.new_page()
        
        try:
            # 访问前端页面
            print("访问前端页面...")
            page.goto(FRONTEND_URL)
            time.sleep(2)
            
            # 点击订单管理
            print("点击订单管理...")
            page.click("text=订单管理")
            time.sleep(2)
            
            # 点击新增订单
            print("点击新增订单...")
            page.click("text=新增订单")
            time.sleep(2)
            
            # 选择业务用户
            print("选择业务用户...")
            page.click("#el-select-1")
            time.sleep(1)
            page.click("#el-select-1 .el-select-dropdown__item")
            time.sleep(2)
            
            # 选择发件人
            print("选择发件人...")
            page.click("#el-select-2")
            time.sleep(1)
            page.click("#el-select-2 .el-select-dropdown__item")
            time.sleep(2)
            
            # 选择收件人
            print("选择收件人...")
            page.click("#el-select-3")
            time.sleep(1)
            page.click("#el-select-3 .el-select-dropdown__item")
            time.sleep(2)
            
            # 填写订单信息
            print("填写订单信息...")
            page.fill("input[placeholder='请输入订单编号']", "TEST-" + str(int(time.time())))
            page.select_option("select", "0")  # 选择订单类型
            page.fill("input[placeholder='请输入货物名称']", "测试货物")
            page.fill("input[placeholder='请输入数量']", "10")
            page.fill("input[placeholder='请输入重量']", "5.5")
            page.fill("input[placeholder='请输入体积']", "2.3")
            page.fill("input[placeholder='请输入总费用']", "100.0")
            page.select_option("select", "0")  # 选择付款方式
            page.select_option("select", "0")  # 选择状态
            time.sleep(2)
            
            # 提交订单
            print("提交订单...")
            page.click("text=确定")
            time.sleep(2)
            
            # 验证订单是否创建成功
            print("验证订单是否创建成功...")
            # 这里可以添加验证逻辑，比如检查是否有成功提示
            
        except Exception as e:
            print(f"测试失败: {e}")
        finally:
            # 关闭浏览器
            browser.close()

# 测试性能
def test_performance():
    """测试前端性能"""
    print("\n开始前端性能测试...")
    
    with sync_playwright() as p:
        # 启动浏览器
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        
        try:
            # 访问前端页面并测量加载时间
            print("测量页面加载时间...")
            start_time = time.time()
            page.goto(FRONTEND_URL)
            end_time = time.time()
            load_time = end_time - start_time
            print(f"页面加载时间: {load_time:.4f} 秒")
            
            # 测量订单页面加载时间
            print("测量订单页面加载时间...")
            start_time = time.time()
            page.click("text=订单管理")
            page.wait_for_load_state("networkidle")
            end_time = time.time()
            load_time = end_time - start_time
            print(f"订单页面加载时间: {load_time:.4f} 秒")
            
        except Exception as e:
            print(f"测试失败: {e}")
        finally:
            # 关闭浏览器
            browser.close()

if __name__ == "__main__":
    test_order_page()
    test_performance()
    print("\nUI测试完成!")

File: .trae/skills/playwright-testing/scripts/ui-test.js
/**
 * UI Testing Script using Playwright
 * Tests frontend pages and user interactions
 */

import { chromium } from '@playwright/test';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const CONFIG = {
  baseUrl: 'http://localhost:3000',
  credentials: {
    customer: { username: 'customer1', password: '123456' },
    driver: { username: 'driver1', password: '123456' },
    network: { username: 'network1', password: '123456' },
    dispatcher: { username: 'dispatcher1', password: '123456' }
  },
  screenshots: {
    dir: path.join(__dirname, '..', 'assets', 'screenshots')
  },
  browser: {
    executablePath: '/Users/jiangxiaochun/Library/Caches/ms-playwright/chromium-1105/chrome-mac/Chromium.app/Contents/MacOS/Chromium'
  }
};

const testResults = {
  success: true,
  steps: [],
  errors: []
};

function logStep(step, content, expected, actual, status = 'pass') {
  const result = {
    step,
    content,
    expected,
    actual,
    status,
    timestamp: new Date().toISOString()
  };
  testResults.steps.push(result);
  console.log(`「${step}；${content}；期望：${expected}；实际：${actual}」`);
  return result;
}

function logError(error) {
  testResults.errors.push(error);
  testResults.success = false;
  console.error(`[ERROR] ${error.message}`);
}

async function takeScreenshot(page, name) {
  try {
    const screenshotPath = path.join(CONFIG.screenshots.dir, `${name}_${Date.now()}.png`);
    await page.screenshot({ path: screenshotPath, fullPage: true });
    console.log(`  Screenshot saved: ${screenshotPath}`);
    return screenshotPath;
  } catch (error) {
    console.error(`  Screenshot failed: ${error.message}`);
    return null;
  }
}

async function testLogin(role = 'customer') {
  console.log(`\n=== Testing ${role} Login ===`);

  const browser = await chromium.launch({ headless: true, executablePath: CONFIG.browser.executablePath });
  const page = await browser.newPage();
  await page.setViewportSize({ width: 1920, height: 1080 });

  try {
    const credentials = CONFIG.credentials[role];

    // Navigate to login page
    await page.goto(`${CONFIG.baseUrl}/login`, {
      waitUntil: 'networkidle',
      timeout: 30000
    });

    logStep('Navigation', 'Open login page', 'Page loaded', 'Success');

    // Fill credentials
    await page.fill('input[type="text"]', credentials.username);
    await page.fill('input[type="password"]', credentials.password);

    logStep('Input', 'Fill credentials', 'Filled', `User: ${credentials.username}`);

    // Click login button
    await page.click('button[type="submit"], .el-button--primary');

    // Wait for navigation
    await page.waitForLoadState('networkidle', { timeout: 15000 });

    logStep('Login', 'Click login button', 'Success', 'Success');

    // Take screenshot
    await takeScreenshot(page, `${role}_home`);

    // Verify we're on home page
    const url = page.url();
    logStep('Verification', 'Check redirect', 'Home page', url.includes('/home') ? 'Home' : url);

    return { browser, page, success: true };

  } catch (error) {
    logError(error);
    await takeScreenshot(page, `${role}_error`);
    await browser.close();
    return { browser: null, page: null, success: false };
  }
}

async function testCustomerOrderPage() {
  console.log('\n=== Testing Customer Order Page ===');

  let browser, page;

  try {
    // Login first
    const loginResult = await testLogin('customer');
    if (!loginResult.success) return false;
    browser = loginResult.browser;
    page = loginResult.page;

    // Navigate to order page
    await page.goto(`${CONFIG.baseUrl}/order`, {
      waitUntil: 'networkidle',
      timeout: 30000
    });

    logStep('Navigation', 'Go to order page', 'Success', 'Success');

    // Take screenshot
    await takeScreenshot(page, 'customer_orders');

    // Check page title
    const title = await page.evaluate(() => {
      const selectors = ['.page-title', '.header-title', 'h1', 'h2'];
      for (const sel of selectors) {
        const el = document.querySelector(sel);
        if (el && el.textContent.trim()) {
          return el.textContent.trim();
        }
      }
      return document.title || 'No title';
    });

    const hasOrderTitle = title.includes('订单') || title.includes('我的订单');
    logStep('Verification', 'Check page title', 'Contains "订单"', title, hasOrderTitle ? 'pass' : 'fail');

    // Check for order table
    const hasTable = await page.evaluate(() => {
      return !!(document.querySelector('.el-table') ||
                document.querySelector('table') ||
                document.querySelector('.order-table'));
    });
    logStep('Verification', 'Check order table', 'Exists', hasTable ? 'Found' : 'Not found', hasTable ? 'pass' : 'fail');

    // Check for "New Order" button
    const hasNewOrderBtn = await page.evaluate(() => {
      const keywords = ['新增订单', '新建订单', '添加订单', '创建订单'];
      const buttons = document.querySelectorAll('button, .el-button, a');
      for (const btn of buttons) {
        const text = btn.textContent.trim();
        if (keywords.some(kw => text.includes(kw))) {
          return true;
        }
      }
      return false;
    });
    logStep('Verification', 'Check new order button', 'Exists', hasNewOrderBtn ? 'Found' : 'Not found', hasNewOrderBtn ? 'pass' : 'fail');

    await browser.close();
    return true;

  } catch (error) {
    logError(error);
    if (browser) await browser.close();
    return false;
  }
}

async function testDriverOrderPage() {
  console.log('\n=== Testing Driver Order Page ===');

  let browser, page;

  try {
    const loginResult = await testLogin('driver');
    if (!loginResult.success) return false;
    browser = loginResult.browser;
    page = loginResult.page;

    await page.goto(`${CONFIG.baseUrl}/driver/order`, {
      waitUntil: 'networkidle',
      timeout: 30000
    });

    logStep('Navigation', 'Go to driver order page', 'Success', 'Success');
    await takeScreenshot(page, 'driver_orders');

    await browser.close();
    return true;

  } catch (error) {
    logError(error);
    if (browser) await browser.close();
    return false;
  }
}

async function testNetworkOrderPage() {
  console.log('\n=== Testing Network Order Page ===');

  let browser, page;

  try {
    const loginResult = await testLogin('network');
    if (!loginResult.success) return false;
    browser = loginResult.browser;
    page = loginResult.page;

    await page.goto(`${CONFIG.baseUrl}/network/order`, {
      waitUntil: 'networkidle',
      timeout: 30000
    });

    logStep('Navigation', 'Go to network order page', 'Success', 'Success');
    await takeScreenshot(page, 'network_orders');

    await browser.close();
    return true;

  } catch (error) {
    logError(error);
    if (browser) await browser.close();
    return false;
  }
}

async function testDispatcherPage() {
  console.log('\n=== Testing Dispatcher Page ===');

  let browser, page;

  try {
    const loginResult = await testLogin('dispatcher');
    if (!loginResult.success) return false;
    browser = loginResult.browser;
    page = loginResult.page;

    await page.goto(`${CONFIG.baseUrl}/dispatch`, {
      waitUntil: 'networkidle',
      timeout: 30000
    });

    logStep('Navigation', 'Go to dispatcher page', 'Success', 'Success');
    await takeScreenshot(page, 'dispatcher');

    await browser.close();
    return true;

  } catch (error) {
    logError(error);
    if (browser) await browser.close();
    return false;
  }
}

async function runAllTests() {
  console.log('=== UI Testing with Playwright ===');
  console.log(`Base URL: ${CONFIG.baseUrl}`);
  console.log(`Screenshots: ${CONFIG.screenshots.dir}`);

  await testCustomerOrderPage();
  await testDriverOrderPage();
  await testNetworkOrderPage();
  await testDispatcherPage();

  console.log('\n=== UI Test Summary ===');
  console.log(`Total steps: ${testResults.steps.length}`);
  console.log(`Passed: ${testResults.steps.filter(s => s.status === 'pass').length}`);
  console.log(`Failed: ${testResults.steps.filter(s => s.status === 'fail').length}`);
  console.log(`Errors: ${testResults.errors.length}`);
  console.log(`\nOverall: ${testResults.success ? 'PASSED' : 'FAILED'}`);

  return testResults;
}

runAllTests().then(results => {
  process.exit(results.success ? 0 : 1);
}).catch(error => {
  console.error('Test execution failed:', error);
  process.exit(1);
});
